include/swift/Parse/Parser.h

```diff
@@ -700,11 +700,58 @@ class Parser {
   bool parseMatchingToken(tok K, SourceLoc &TokLoc, Diag<> ErrorDiag,
                           SourceLoc OtherLoc);
 
+  /// Tracking IfConfigDecl for conditionals in collections for AST dump.
+  /// Since there is no place to directly represent conditionals in the
+  /// current AST collection model, a map is generated from the location
+  /// of actual data values to the conditionals that directly preceed them.
+  /// This is transferred to the CollectionExpr when the collection completes
+  /// and is used by the ATSDumper to provide a repsentation of the conditional.
```

Comment:
If there other important consumers they will need to be adapted like ASTDumper and Syntax. What do you have in mind?

Reply:
`FormatWalker` in `lib/IDE/Formatting.cpp` walks into `IfConfigDecl`. We need to perform code formatting even for inactive `#if` directive blocks.
Eventually, code-formatting can be done by using `libSyntax`, but currently, we are still using `AST`.

Reply:
Hmmm. While all elements, active and inactive syntax high OK in the IDE, I’m not sure I can give you a solution on auto-indentation inside the scope of this PR. This is a difficult problem. I’ve updated the comment.

Reply:
There is even a FIXME in lib/IDE/Formatting.cpp 
```
        // FIXME: We are going to revisit the behavior and the indentation we
        // want for dictionary/array literals.
```
Better left until #sourceLocation is sorted out and it can be tackled separately.